src/commands.ts

```diff
@@ -8,42 +8,79 @@ import { Remote } from "./remote"
 import { Storage } from "./storage"
 import { OpenableTreeItem } from "./workspacesProvider"
 
-// maybeAskUrl asks the user for the URL if it was not provided and normalizes
-// the returned URL.
-export async function maybeAskUrl(
-  providedUrl: string | undefined | null,
-  lastUsedUrl?: string,
-): Promise<string | undefined> {
-  let url =
-    providedUrl ||
-    (await vscode.window.showInputBox({
-      title: "Coder URL",
-      prompt: "Enter the URL of your Coder deployment.",
-      placeHolder: "https://example.coder.com",
-      value: lastUsedUrl,
-    }))
-  if (!url) {
-    return undefined
-  }
-  if (!url.startsWith("http://") && !url.startsWith("https://")) {
-    // Default to HTTPS if not provided!
-    // https://github.com/coder/vscode-coder/issues/44
-    url = "https://" + url
-  }
-  while (url.endsWith("/")) {
-    url = url.substring(0, url.length - 1)
-  }
-  return url
-}
-
 export class Commands {
   public constructor(
     private readonly vscodeProposed: typeof vscode,
     private readonly storage: Storage,
   ) {}
 
+  /**
+   * Ask the user for the URL, letting them choose from a list of recent URLs or
+   * CODER_URL or enter a new one.  Undefined means the user aborted.
+   */
+  private askURL(selection?: string): Promise<string | undefined> {
+    const quickPick = vscode.window.createQuickPick()
+    quickPick.value = selection || process.env.CODER_URL || ""
```

Comment:
Do we need to validate the user-entered URL in any way?

Reply:
Good question, thinking about it we could validate that the characters are allowable in a URL and we could also maybe check if the protocol is not something unexpected like `file:///` although I think it might already be enough that these "bad" URLs will fail in the login step before we store the URL.

Reply:
Ahh that seems fine then!